tests/integration/suite/daprd/resources/namespace.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	rtv1 "github.com/dapr/dapr/pkg/proto/runtime/v1"
+	"github.com/dapr/dapr/tests/integration/framework"
+	"github.com/dapr/dapr/tests/integration/framework/process/daprd"
+	"github.com/dapr/dapr/tests/integration/framework/process/exec"
+	"github.com/dapr/dapr/tests/integration/suite"
+)
+
+func init() {
+	suite.Register(new(namespace))
+}
+
+// namespace ensures that the component disk loader does not load components
+// from different namespaces.
+type namespace struct {
+	daprd *daprd.Daprd
+}
+
+func (n *namespace) Setup(t *testing.T) []framework.Option {
+	n.daprd = daprd.New(t, daprd.WithResourceFiles(`
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: abc
+spec:
+  type: state.in-memory
+  version: v1
+---
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: ghi
+  namespace: notmynamespace
+spec:
+  type: state.in-memory
+  version: v1
+---
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: abc
```

Comment:
nitpick change name to something else.

Reply:
It might be clearer if you moved up the `name: abc` w/ the `namespace: notmynamespace` right below the first one with `name: abc` but no namespace to prove your point in a more reader friendly manner, otherwise we see `name: ghi` which is next, but not sequentially next in the alphabet and `def` later which is just out of order and then takes more time to look at to figure out what you are trying to show with these examples.

Reply:
Done